CMakeLists.txt

```diff
@@ -74,8 +75,11 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wconversion")
 if(APPLE AND DEFINED ENV{HOMEBREW_PREFIX})
     # Used for dlopen finding dylibs installed by homebrew, reversed for future usage
     # `/opt/homebrew/lib` is not on dlopen search path by default
-    set(CMAKE_EXE_LINKER_FLAGS "-Wl,-rpath,$ENV{HOMEBREW_PREFIX}/lib")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath,$ENV{HOMEBREW_PREFIX}/lib")
 endif()
+set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined")
+set(CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -rdynamic -fno-omit-frame-pointer -fsanitize=address -fsanitize=undefined")
```

Comment:
https://github.com/google/sanitizers/wiki/AddressSanitizer

For example

https://github.com/LinusDierheimer/fastfetch/blob/47c9fb9f6cf6bb2fe7f15dca23fa8adfe3a42b57/src/util/FFlist.c#L27

If we did write out of buffer it could be detected by `-fsanitize=address`

`-fsanitize=leak` will be much more useful however it isn't supported on macOS ( can be enabled for linux though )